### ISSUE TITLE
Istio non-default certs are not being read correctly

### ISSUE THREAD

**Issue by rickducott** [solo-io/supergloo#100]
Leftover work from #98

**Comment by rickducott** [comment]
The issue is how we serialize the certificate. Here's an example of what we serialize to the secret: 
```
  root-cert.pem: fC0KICAtLS0tLUJFR0lOIENFUlRJRklDQVRFLS0tLS0KICBNSUlEN1RDQ0F0V2dBd0lCQWdJSkFPSVJEaE9jeHN4Nk1BMEdDU3FHU0liM0RRRUJDd1VBTUlHTE1Rc3dDUVlECiAgVlFRR0V3SlZVekVUTUJFR0ExVUVDQXdLUTJGc2FXWnZjbTVwWVRFU01CQUdBMVVFQnd3SlUzVnVibmwyWVd4bAogIE1RNHdEQVlEVlFRS0RBVkpjM1JwYnpFTk1Bc0dBMVVFQ3d3RVZHVnpkREVRTUE0R0ExVUVBd3dIVW05dmRDQkQKICBRVEVpTUNBR0NTcUdTSWIzRFFFSkFSWVRkR1Z6ZEhKdmIzUmpZVUJwYzNScGJ5NXBiekFnRncweE9EQXhNalF4CiAgT1RFMU5URmFHQTh5TVRFM01USXpNVEU1TVRVMU1Wb3dnWXN4Q3pBSkJnTlZCQVlUQWxWVE1STXdFUVlEVlFRSQogIERBcERZV3hwWm05eWJtbGhNUkl3RUFZRFZRUUhEQWxUZFc1dWVYWmhiR1V4RGpBTUJnTlZCQW9NQlVsemRHbHYKICBNUTB3Q3dZRFZRUUxEQVJVWlhOME1SQXdEZ1lEVlFRRERBZFNiMjkwSUVOQk1TSXdJQVlKS29aSWh2Y05BUWtCCiAgRmhOMFpYTjBjbTl2ZEdOaFFHbHpkR2x2TG1sdk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQgogIENnS0NBUUVBMzh1RWZBYXR6UVlxYmFMb3UxbnhKMzQ4VnlOenVtWU1tRER0NXBiTFlSckNvMnBTM2tpMVpWRE4KICA4eXhJRU5KRmtwS3c5VWN0VEdkYk5HdUdDaVNEUDd1cUY2QmlWbitYS0FVLzNwblBGQmJUZDBTMzNOcWJERVF1CiAgSVlyYUhTbC90U2s1ckFSYkMxRHJRUmRaNm5ZRDJLcmFwQzRnMFhialk2UHU1bDR5N0tuRndTdW5ucDl1cXBadwogIHVFUnYvQmd1bUo1UWxTZVNlQ21obkRoTHhvb0c4dzV0QzJ5VnIxeURwc09IR2ltUC9tYzhDZHM0VjB6ZkloUXYKICBZemZJSHBoaEU5REtqbW5qQllMT2RqNGF5Y3Y0NGpIbk9HYyt3dkExSnFzbDYwdDN3Z21zK3pKVGlXd0FCTGR3CiAgemdNQWE3eXhMeW9WMCtQaVZRdWQ2ays4Wm9JRmN3SURBUUFCbzFBd1RqQWRCZ05WSFE0RUZnUVVPVVlHdFV5aAogIGV1eE80bEdlNE9wMXk4TlZvYWd3SHdZRFZSMGpCQmd3Rm9BVU9VWUd0VXloZXV4TzRsR2U0T3AxeThOVm9hZ3cKICBEQVlEVlIwVEJBVXdBd0VCL3pBTkJna3Foa2lHOXcwQkFRc0ZBQU9DQVFFQU5YTHlmQXM3SjlybUJhbUdKdlBaCiAgbHR4MzkwV3h6ekxGUXNCUkFhSDZyZ2VpcEJxM2RSOXFFakF3YjZCVEYrUk9tdFF6WCtmanN0Q1JySnhDdG85VwogIHRDOEt2WFRkUmZJamZDQ1pqaHRJT0JLcVJ4RTRLSlYvUkJmdjl4RDVseWp0Q1BDUWwzSWE2TVNmNDJOK2FiQUsKICBXQ2RVNktDb2pBOFdCOVloU0N6emEzYVFiUFR6ZDI2T0MvSmJsSnBWZ3R1czVmOElMekNzeitwYk1pbWdUa2h5CiAgQXVoWVJwcEphUTI0QVBpanNFQzkrR0lhVktQZzVJd1dyb2lQb2orUVhOcHNodXZxVlFRWHZHYVJpcTR6b1NueAogIHhBSnordzh0anJEV2NmODI2Vk4xNElMKy9DbXFsZy9ySWZCNUNIZHdWSWZXd3B1R0I2NnEvVWlQZWdaTU5zOGEKICAzZz09CiAgLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```

When you base64 decode that string, it looks like this:
```
|-
  -----BEGIN CERTIFICATE-----
  MIID7TCCAtWgAwIBAgIJAOIRDhOcxsx6MA0GCSqGSIb3DQEBCwUAMIGLMQswCQYD
  VQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTESMBAGA1UEBwwJU3Vubnl2YWxl
  MQ4wDAYDVQQKDAVJc3RpbzENMAsGA1UECwwEVGVzdDEQMA4GA1UEAwwHUm9vdCBD
  QTEiMCAGCSqGSIb3DQEJARYTdGVzdHJvb3RjYUBpc3Rpby5pbzAgFw0xODAxMjQx
  OTE1NTFaGA8yMTE3MTIzMTE5MTU1MVowgYsxCzAJBgNVBAYTAlVTMRMwEQYDVQQI
  DApDYWxpZm9ybmlhMRIwEAYDVQQHDAlTdW5ueXZhbGUxDjAMBgNVBAoMBUlzdGlv
  MQ0wCwYDVQQLDARUZXN0MRAwDgYDVQQDDAdSb290IENBMSIwIAYJKoZIhvcNAQkB
  FhN0ZXN0cm9vdGNhQGlzdGlvLmlvMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
  CgKCAQEA38uEfAatzQYqbaLou1nxJ348VyNzumYMmDDt5pbLYRrCo2pS3ki1ZVDN
  8yxIENJFkpKw9UctTGdbNGuGCiSDP7uqF6BiVn+XKAU/3pnPFBbTd0S33NqbDEQu
  IYraHSl/tSk5rARbC1DrQRdZ6nYD2KrapC4g0XbjY6Pu5l4y7KnFwSunnp9uqpZw
  uERv/BgumJ5QlSeSeCmhnDhLxooG8w5tC2yVr1yDpsOHGimP/mc8Cds4V0zfIhQv
  YzfIHphhE9DKjmnjBYLOdj4aycv44jHnOGc+wvA1Jqsl60t3wgms+zJTiWwABLdw
  zgMAa7yxLyoV0+PiVQud6k+8ZoIFcwIDAQABo1AwTjAdBgNVHQ4EFgQUOUYGtUyh
  euxO4lGe4Op1y8NVoagwHwYDVR0jBBgwFoAUOUYGtUyheuxO4lGe4Op1y8NVoagw
  DAYDVR0TBAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEANXLyfAs7J9rmBamGJvPZ
  ltx390WxzzLFQsBRAaH6rgeipBq3dR9qEjAwb6BTF+ROmtQzX+fjstCRrJxCto9W
  tC8KvXTdRfIjfCCZjhtIOBKqRxE4KJV/RBfv9xD5lyjtCPCQl3Ia6MSf42N+abAK
  WCdU6KCojA8WB9YhSCzza3aQbPTzd26OC/JblJpVgtus5f8ILzCsz+pbMimgTkhy
  AuhYRppJaQ24APijsEC9+GIaVKPg5IwWroiPoj+QXNpshuvqVQQXvGaRiq4zoSnx
  xAJz+w8tjrDWcf826VN14IL+/Cmqlg/rIfB5CHdwVIfWwpuGB66q/UiPegZMNs8a
  3g==
```

When you save a secret via Kubectl, it looks like this:
```
  root-cert.pem: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUQ3VENDQXRXZ0F3SUJBZ0lKQU9JUkRoT2N4c3g2TUEwR0NTcUdTSWIzRFFFQkN3VUFNSUdMTVFzd0NRWUQKVlFRR0V3SlZVekVUTUJFR0ExVUVDQXdLUTJGc2FXWnZjbTVwWVRFU01CQUdBMVVFQnd3SlUzVnVibmwyWVd4bApNUTR3REFZRFZRUUtEQVZKYzNScGJ6RU5NQXNHQTFVRUN3d0VWR1Z6ZERFUU1BNEdBMVVFQXd3SFVtOXZkQ0JEClFURWlNQ0FHQ1NxR1NJYjNEUUVKQVJZVGRHVnpkSEp2YjNSallVQnBjM1JwYnk1cGJ6QWdGdzB4T0RBeE1qUXgKT1RFMU5URmFHQTh5TVRFM01USXpNVEU1TVRVMU1Wb3dnWXN4Q3pBSkJnTlZCQVlUQWxWVE1STXdFUVlEVlFRSQpEQXBEWVd4cFptOXlibWxoTVJJd0VBWURWUVFIREFsVGRXNXVlWFpoYkdVeERqQU1CZ05WQkFvTUJVbHpkR2x2Ck1RMHdDd1lEVlFRTERBUlVaWE4wTVJBd0RnWURWUVFEREFkU2IyOTBJRU5CTVNJd0lBWUpLb1pJaHZjTkFRa0IKRmhOMFpYTjBjbTl2ZEdOaFFHbHpkR2x2TG1sdk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQgpDZ0tDQVFFQTM4dUVmQWF0elFZcWJhTG91MW54SjM0OFZ5Tnp1bVlNbUREdDVwYkxZUnJDbzJwUzNraTFaVkROCjh5eElFTkpGa3BLdzlVY3RUR2RiTkd1R0NpU0RQN3VxRjZCaVZuK1hLQVUvM3BuUEZCYlRkMFMzM05xYkRFUXUKSVlyYUhTbC90U2s1ckFSYkMxRHJRUmRaNm5ZRDJLcmFwQzRnMFhialk2UHU1bDR5N0tuRndTdW5ucDl1cXBadwp1RVJ2L0JndW1KNVFsU2VTZUNtaG5EaEx4b29HOHc1dEMyeVZyMXlEcHNPSEdpbVAvbWM4Q2RzNFYwemZJaFF2Cll6ZklIcGhoRTlES2ptbmpCWUxPZGo0YXljdjQ0akhuT0djK3d2QTFKcXNsNjB0M3dnbXMrekpUaVd3QUJMZHcKemdNQWE3eXhMeW9WMCtQaVZRdWQ2ays4Wm9JRmN3SURBUUFCbzFBd1RqQWRCZ05WSFE0RUZnUVVPVVlHdFV5aApldXhPNGxHZTRPcDF5OE5Wb2Fnd0h3WURWUjBqQkJnd0ZvQVVPVVlHdFV5aGV1eE80bEdlNE9wMXk4TlZvYWd3CkRBWURWUjBUQkFVd0F3RUIvekFOQmdrcWhraUc5dzBCQVFzRkFBT0NBUUVBTlhMeWZBczdKOXJtQmFtR0p2UFoKbHR4MzkwV3h6ekxGUXNCUkFhSDZyZ2VpcEJxM2RSOXFFakF3YjZCVEYrUk9tdFF6WCtmanN0Q1JySnhDdG85Vwp0QzhLdlhUZFJmSWpmQ0Naamh0SU9CS3FSeEU0S0pWL1JCZnY5eEQ1bHlqdENQQ1FsM0lhNk1TZjQyTithYkFLCldDZFU2S0NvakE4V0I5WWhTQ3p6YTNhUWJQVHpkMjZPQy9KYmxKcFZndHVzNWY4SUx6Q3N6K3BiTWltZ1RraHkKQXVoWVJwcEphUTI0QVBpanNFQzkrR0lhVktQZzVJd1dyb2lQb2orUVhOcHNodXZxVlFRWHZHYVJpcTR6b1NueAp4QUp6K3c4dGpyRFdjZjgyNlZOMTRJTCsvQ21xbGcvcklmQjVDSGR3VklmV3dwdUdCNjZxL1VpUGVnWk1OczhhCjNnPT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```

Which decodes to:
```
-----BEGIN CERTIFICATE-----
MIID7TCCAtWgAwIBAgIJAOIRDhOcxsx6MA0GCSqGSIb3DQEBCwUAMIGLMQswCQYD
VQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTESMBAGA1UEBwwJU3Vubnl2YWxl
MQ4wDAYDVQQKDAVJc3RpbzENMAsGA1UECwwEVGVzdDEQMA4GA1UEAwwHUm9vdCBD
QTEiMCAGCSqGSIb3DQEJARYTdGVzdHJvb3RjYUBpc3Rpby5pbzAgFw0xODAxMjQx
OTE1NTFaGA8yMTE3MTIzMTE5MTU1MVowgYsxCzAJBgNVBAYTAlVTMRMwEQYDVQQI
DApDYWxpZm9ybmlhMRIwEAYDVQQHDAlTdW5ueXZhbGUxDjAMBgNVBAoMBUlzdGlv
MQ0wCwYDVQQLDARUZXN0MRAwDgYDVQQDDAdSb290IENBMSIwIAYJKoZIhvcNAQkB
FhN0ZXN0cm9vdGNhQGlzdGlvLmlvMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
CgKCAQEA38uEfAatzQYqbaLou1nxJ348VyNzumYMmDDt5pbLYRrCo2pS3ki1ZVDN
8yxIENJFkpKw9UctTGdbNGuGCiSDP7uqF6BiVn+XKAU/3pnPFBbTd0S33NqbDEQu
IYraHSl/tSk5rARbC1DrQRdZ6nYD2KrapC4g0XbjY6Pu5l4y7KnFwSunnp9uqpZw
uERv/BgumJ5QlSeSeCmhnDhLxooG8w5tC2yVr1yDpsOHGimP/mc8Cds4V0zfIhQv
YzfIHphhE9DKjmnjBYLOdj4aycv44jHnOGc+wvA1Jqsl60t3wgms+zJTiWwABLdw
zgMAa7yxLyoV0+PiVQud6k+8ZoIFcwIDAQABo1AwTjAdBgNVHQ4EFgQUOUYGtUyh
euxO4lGe4Op1y8NVoagwHwYDVR0jBBgwFoAUOUYGtUyheuxO4lGe4Op1y8NVoagw
DAYDVR0TBAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEANXLyfAs7J9rmBamGJvPZ
ltx390WxzzLFQsBRAaH6rgeipBq3dR9qEjAwb6BTF+ROmtQzX+fjstCRrJxCto9W
tC8KvXTdRfIjfCCZjhtIOBKqRxE4KJV/RBfv9xD5lyjtCPCQl3Ia6MSf42N+abAK
WCdU6KCojA8WB9YhSCzza3aQbPTzd26OC/JblJpVgtus5f8ILzCsz+pbMimgTkhy
AuhYRppJaQ24APijsEC9+GIaVKPg5IwWroiPoj+QXNpshuvqVQQXvGaRiq4zoSnx
xAJz+w8tjrDWcf826VN14IL+/Cmqlg/rIfB5CHdwVIfWwpuGB66q/UiPegZMNs8a
3g==
```

**Comment by rickducott** [comment]
Have confirmed that once we resolve the two related solo-kit bugs, custom certs with istio work. Specifically, in order to use the solo kit secret client we need:
* Default / empty values should be written into the secret as the empty string ("")
* Non-empty values should be written as strings, not as yaml-encoded strings

**Comment by rickducott** [comment]
@ilackarms fixed this in solo-kit dbdc438480bf94648fa493d07e62341219f8167f, testing out the fix now
